### PR TITLE
Fix ng2 jhiPrefix compile errors

### PR DESCRIPTION
**Pull request by mehmet-aslan**
This only fixes compile errors if jhiPrefix is different from default. There could be more things to do to complete ng2-jhiPrefix support.

Also fixed URL in metrics.service.

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/shared/language/_jhi-translate.directive.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { Component, Input } from '@angular/core';
     selector: '[jhi-translate]',
     template: '<span [innerHTML]="jhiTranslate | translate:translateValues"></span>'
 })
-export class JhiTranslate {
+export class <%=jhiPrefixCapitalized%>Translate {
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/language/_jhi-translate.directive.ts]: this should be `Jhi` always as the directive will always be `jhi-translate`

File: generators/client/templates/angular/src/main/webapp/app/shared/component/_jhi-item-count.component.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { Component, Input } from '@angular/core';
             of {{total}} items.
         </div>`
 })
-export class JhiItemCountComponent {
+export class <%=jhiPrefixCapitalized%>ItemCountComponent {
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/component/_jhi-item-count.component.ts]: this should be `Jhi` always as the directive will always be `jhi-item-count`

File: generators/client/templates/angular/src/main/webapp/app/shared/alert/_alert.component.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import { AlertService } from './alert.service';
             </div>
         </div>`
 })
-export class JhiAlertComponent implements OnInit, OnDestroy {
+export class <%=jhiPrefixCapitalized%>AlertComponent implements OnInit, OnDestroy {
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/alert/_alert.component.ts]: this should be `Jhi` always as the directive will always be `jhi-alert`

File: generators/client/templates/angular/src/main/webapp/app/shared/alert/_alert-error.component.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ import { Subscription } from 'rxjs/Rx';
             </div>
         </div>`
 })
-export class JhiAlertErrorComponent implements OnDestroy {
+export class <%=jhiPrefixCapitalized%>AlertErrorComponent implements OnDestroy {
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/alert/_alert-error.component.ts]: this should be `Jhi` always as the directive will always be `jhi-*`

File: generators/client/templates/angular/src/main/webapp/app/shared/_shared-common.module.ts [hunk 5]
```diff
@@ -20,10 +20,10 @@ import {
     MaxbytesValidator,
     MinbytesValidator,
     ShowValidationDirective,
-    JhiItemCountComponent,
+    <%=jhiPrefixCapitalized%>ItemCountComponent,
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/_shared-common.module.ts]: These shoudnt be changed this should be `Jhi` always as the directive will always be `jhi-*`

File: generators/client/templates/angular/src/main/webapp/app/shared/language/_jhi-translate.directive.ts (hunk 1 above)
Comment (mehmet-aslan) [inline generators/client/templates/angular/src/main/webapp/app/shared/language/_jhi-translate.directive.ts]: This gives compile error at [shared-common.module.ts](https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/app/shared/_shared-common.module.ts#L14). This needs to be changed to `JhiTranslate` in this case.

Comment (mehmet-aslan) [inline generators/client/templates/angular/src/main/webapp/app/shared/language/_jhi-translate.directive.ts]: So `JhiMissingTranslationHandler` and `JhiLanguageService` classes could also be changed from `jhiPrefix` to `Jhi` to achieve this.

I updated PR without doing this to only fix compile errors. They could be changed with another PR if needed.
